runtime/compiler/compile/J9Compilation.cpp

```diff
@@ -665,11 +667,12 @@ J9::Compilation::canAllocateInline(TR::Node* node, TR_OpaqueClassBlock* &classIn
       {
       classRef      = node->getSecondChild();
 
-      // In the case of dynamic array allocation, return 0 indicating variable dynamic array allocation
+      // In the case of dynamic array allocation, return 0 indicating variable dynamic array allocation,
+      // unless value types are enabled, in which case return -1 to prevent inline allocation
       if (classRef->getOpCodeValue() != TR::loadaddr)
          {
          classInfo = NULL;
-         return 0;
+         return areValueTypesEnabled ? -1 : 0;
```

Comment:
This is a pretty big thing since the dynamic allocations won't work - can we have a debug counter and a trace statement? Ultimately we would want to check at runtime if we could do the inline allocation otherwise call out, but this is fine as a first step for functional correctness.

Reply:
Good point!  Added a new debug counter in this revised commit:  https://github.com/eclipse/openj9/pull/9139/commits/b27c9e6af6c7645dea2b198333184faa14b3df04